kernel/of/fdt.c

```diff
@@ -285,7 +414,13 @@ fdt_string_from_offset(struct fdt* fdt, size_t offset)
 const char*
 fdt_node_name(struct fdt_node* node)
 {
+#ifndef FDT_DEBUG_DEBUG_NAMELESS_NODES
     return node->unit_name;
+#else
+    // If the first character is the terminator, replace it with a debug name
+    // NOTE: different nodes can appear to have the same name with this scheme
+    return *(char*)node->unit_name != '\0' ? node->unit_name : "{DEBUG: Nameless Node}";
```

Comment:
is `unit_name` not already a `char*`?

Reply:
Technically no, it's a `char[]` but I'm pretty much certain dereferencing an array like `*unit_name` is allowed, I just didn't feel like double checking it is. (We could also just do `unit_name[0]`) all of them are equivalent afaik

Reply:
I can change it if it really bothers you but it doesn't bother me.

Reply:
Ya actually I'd do `unit_name[0]` it's clear what you mean then